python/ray/serve/pipeline/deployment_node.py

```diff
@@ -1,34 +1,51 @@
 from typing import Any, Dict, Optional, List, Tuple, Union
 
 from ray.experimental.dag import DAGNode, InputNode
-from ray.serve.api import Deployment
+from ray.experimental.dag.py_obj_scanner import _PyObjScanner
```

Comment:
this should be a private API, if we need to use it here it should be made a public (developer) interface. probably it makes sense that this is a method that we inherit by subclassing `DAGNode` ?